### PR TITLE
Add submission validation

### PR DESCRIPTION
**Pull request by jmthibault79**
Includes some refactoring of the Create Submission service in order to make these pieces available to Validate Submission.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala [hunk 1]
```diff
@@ -77,9 +77,15 @@ case class Submission(
   status: SubmissionStatus
 )
 
-object ExecutionJsonSupport extends JsonSupport {
+// Validation of a Submission
+case class SubmissionValidation(
+  validatedInputs: Seq[Entity],
+  failedInputs: Seq[String],
+  runnableWorkflows: Seq[Map[String, String]],    // not a single map because keys are not unique
```
Comment (tedsharpe) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: Shouldn't this just be a sequence of pairs?

File: src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala [hunk 2]
```diff
@@ -628,42 +621,47 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
...
-                    "are not the same type as the method configuration requires")
-                }
+  def validateSubmission(workspaceName: WorkspaceName, submission: SubmissionRequest): PerRequestMessage =
+    dataSource inTransaction { txn =>
+      withWorkspaceContextAndPermissions(workspaceName, GCSAccessLevel.Write, txn) { workspaceContext =>
+        withMethodConfig(workspaceContext, submission.methodConfigurationNamespace, submission.methodConfigurationName, txn) { methodConfig =>
+          withEntity(workspaceContext, submission.entityType, submission.entityName, txn) { entity =>
+            withMethod(workspaceContext, methodConfig.methodRepoMethod.methodNamespace, methodConfig.methodRepoMethod.methodName, methodConfig.methodRepoMethod.methodVersion, userInfo.authCookie) { agoraEntity =>
+              withWdl(agoraEntity) { wdl =>
+                val (validatedInputs, failedInputs) = validateSubmissionEntities(submission, workspaceContext, entity, methodConfig.rootEntityType)
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala]: I am getting confused by naming... validatedInputs are really the workflow entities, correct?

**Comment by jmthibault79** [comment]
After discussion with Doug, I'm going to change how I handle runnable/unrunnable Workflows.

**Comment by mattsooknah** [comment]
Closing for now until proposed changes have been made.

**Pull request closed without merging**